backend/src/main/kotlin/fr/gouv/cacem/monitorenv/infrastructure/database/repositories/JpaMissionRepository.kt

```diff
@@ -62,31 +59,18 @@ class JpaMissionRepository(
 
     @Transactional
     override fun save(mission: MissionEntity): MissionDTO {
-        return try {
-            // Extract all control units resources unique baseIds
-            val uniqueBaseIds = mission.controlUnits.flatMap { controlUnit ->
-                controlUnit.resources.map { it.baseId }
-            }.distinct()
-            // Fetch all of them as models
-            val baseModels = dbBaseRepository.findAllById(uniqueBaseIds).map { BaseModel.fromFullBase(it) }
-            // Create a `[baseId] → BaseModel` map
-            val baseModelMap = baseModels.associateBy { requireNotNull(it.id) }
+        // Extract all control units resources unique control unit resource IDs
+        val uniqueControlUnitResourceIds = mission.controlUnits.flatMap { controlUnit ->
+            controlUnit.resources.map { it.id }
+        }.distinct()
+        // Fetch all of them as models
+        val controlUnitResourceModels = dbControlUnitResourceRepository.findAllById(uniqueControlUnitResourceIds)
```

Comment:
Est ce qu'on peut utiliser plutot `getReferenceById`pour éviter les select vers la db ?

Reply:
Je garde ton comment ouvert ici avant de merger, comme je l'avais pour certains avec Loup sur la première PR pour ne pas le perdre de l'historique.

Comme ce code n'est pas amené à disparaître (une fois les nouvelles unités déployées, il faudra retirer de fait les unit controls legacy sur Env et sur Fish pour basculer les missions vers l'API /v2), le refacto risque d'être du temps perdu.

Par contre, je garde la référence concernant ta remarque sur l'utilisation `getReferenceById()`. J'essaie de comprendre leur apport. I me semble qu'elles fonctionnent comme `LAZY` (qui d'ailleurs semble inactif sur nos repos ([1](https://stackoverflow.com/questions/62116317/why-does-fetchtype-lazy-not-work-in-hibernate-jpa/62200168#62200168), [2](https://github.com/spring-guides/tut-spring-boot-kotlin#persistence-with-jpa)), en ne fetchant que si la référence est utilisée. Mais la référence est forcément utilisée ici, non ? À moins que cela soit simplement pour avoir un code plus clean et conventionnel ?

Reply:
Pour sauvegarder la mission, on n'a besoin que des identifiants des ressources de controle (qui viennent de la requête)  mais on n'a pas besoin d'autre chose il me semble. GetReferenceById permet cela. Tu peux voir un exemple d'utilisation dans `JpaReportingRepository` pour voir si c'est similaire ou non.

Sinon le `LAZY` fonctionne bien: j'ai dû passer un certain nombre de relations en `EAGER` justement.
(On utilise le plugin allopen)